unpack_action.go

```diff
@@ -12,12 +13,45 @@ type UnpackAction struct {
 	File        string
 }
 
+func tarOptions(compression string) string {
+	unpackTarOpts := map[string]string{
+		"gz":    "-z",
+		"bzip2": "-j",
+		"xz":    "-J",
+	} // Trying to guess all other supported formats
+
+	return unpackTarOpts[compression]
+}
+
+func UnpackTarArchive(infile, destination, compression string, options ...string) error {
```

Comment:
todo for later, move this into some utility file so that we don't have actiosn calling helpers from other actions :)